### ISSUE TITLE
[java] CloseResource false positive on non-SQL classes called Statement

### ISSUE THREAD

**Issue by ThrawnCA** [pmd/pmd#1076]
**Affects PMD Version:** 5.2.3 and higher

**Rule:** [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

**Description:**

Non-SQL classes named "Statement" trigger the `CloseResource` rule even though it should not apply to them.

However, this does not always fire, and the circumstances that make it succeed or fail are unclear.

**Code Sample demonstrating the issue:**

```
    @Test
    public void shouldReissueReceiptWhenReceiptForStatementExists() {
        final Statement paidStatement = StatementSteps.createPaidStatement();
        ...
    }
```

**Running PMD through:** Maven

**Comment by adangel** [comment]
> Non-SQL classes named "Statement" 

This sounds like a typeresolution / incomplete auxclasspath issue.

**Comment by adangel** [comment]
With a correct auxclasspath setting, I can't reproduce this with the current PMD version. So I'm closing this.
If you still have this issue, feel free to reopen it.
